qlib/Mapper.qm

```diff
@@ -980,8 +1014,8 @@ Mapper mapv(DataMap);
             if (m.code) {
                 try {
                     v = do_list
-                        ? (map m.code($1, getrec(rec, $#)), v)
-                        : m.code(v, rec);
+                        ? (map m.code($1, getrec(rec + crec, $#)), v)
```

Comment:
all of these hash concatenation operations should be done once externally to this method call instead of for each field (potentially more than once for each field as well)